### PR TITLE
Changing to work correctly with numpad 0.

### PR DESCRIPTION
**Pull request by jhonatandarosa**
Numpad 0 returns keycode 96, wich is mapped `_MAP` variable. 
Since 0 is false, it does not recognize correctly. For example, pressing ctrl + numpad 0 will recognize ctrl+`

### CODE REVIEW

**Comment by ccampbell** [comment]
Can you please provide a unit test to demonstrate the broken/fixed behavior

**Comment by jhonatandarosa** [comment]
```javascript
it('ctrl+0 not firing when ctrl+ numpad 0 pressed', function() {
    var spy = sinon.spy();

    Mousetrap.bind('ctrl+0', spy);

    // numpad 0 keycode 
    KeyEvent.simulate(96, 96, ['ctrl']);

    expect(spy.callCount).to.equal(1, 'callback should fire once');
    expect(spy.args[0][0]).to.be.an.instanceOf(Event, 'first argument should be Event');
    expect(spy.args[0][1]).to.equal('ctrl+0', 'second argument should be key combo');
});
```

**Comment by tshinnic** [comment]
TL;DR  There's a simpler fix for the original oversight:
```
    154c154
    <         _MAP[i + 96] = i;
    ---
    >         _MAP[i + 96] = i.toString();
```

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -


A key point here is that the use of `"ctrl+0"` forces Mousetrap to use the `keydown` event for detection.  And so the keyCode to 'normal' character translation is needed via `_MAP`.  But for Numpad0 the _MAP[96] value is zero, a troublesome value.

These lines https://github.com/ccampbell/mousetrap/blob/master/mousetrap.js#L150-L155
```
    /**
     * loop through to map numbers on the numeric keypad
     */
    for (i = 0; i <= 9; ++i) {
        _MAP[i + 96] = i;
    }
```
say keycode 96 is `_MAP[96] === 0` or y'know `false`, and so
```
        // for non keypress events the special maps are needed
        if (_MAP[e.which]) {
```
rejects a valid `e.keyCode` / `e.which` value of 96 and doesn't translate.  Worse, even if that is forced to translate using the PR changes, the next stage at https://github.com/ccampbell/mousetrap/blob/master/mousetrap.js#L715-L718
```
            // no character found then stop
            if (!character) {
                return;
```
will reject a character value zero.

So the PR proposed changes fixed both the above condition tests to allow a defined but zero character value.  (I don't think the middle change at line 207 was needed)  I tested changing these two conditionals and indeed the problem was fixed.  


But during testing I noted that only for the numpad keys would the character values from `_characterFromEvent(e)` be returned as typeof 'number'.  In all other interactions this would be a 'string'.  Oh, boo, I says, if only the values were strings ...  And changing that one line mentioned at top, without any of the other changes, also fixes the described problem that `Numpad0` is not matched by `'ctrl+0'`.

.
The tested problem call was `.bind('ctrl+0',...` checking against both normal '0' and keypad '0' keys.

Details of events for reference  (captured from Chrome):
```
 keydown:  charCode=0       keyCode=48("0")  which=48("0")  code="Digit0"  keyIdentifier="U+0030" location=0 
 keypress: charCode=48("0") keyCode=48("0")  which=48("0")  code="Digit0"  keyIdentifier="U+0030" location=0 
 keyup:    charCode=0       keyCode=48("0")  which=48("0")  code="Digit0"  keyIdentifier="U+0030" location=0 

 keydown:  charCode=0       keyCode=96("`")  which=96("`")  code="Numpad0" keyIdentifier="U+0060" location=3 
 keypress: charCode=48("0") keyCode=48("0")  which=48("0")  code="Numpad0" keyIdentifier="U+0030" location=3 
 keyup:    charCode=0       keyCode=96("`")  which=96("`")  code="Numpad0" keyIdentifier="U+0060" location=3 

 keydown:  charCode=0       keyCode=192("À") which=192("À") code="Backquote" keyIdentifier="U+00C0" location=0 
 keypress: charCode=96("`") keyCode=96("`")  which=96("`")  code="Backquote" keyIdentifier="U+0060" location=0 
 keyup:    charCode=0       keyCode=192("À") which=192("À") code="Backquote" keyIdentifier="U+00C0" location=0 
```
